ee/hogai/graph/memory/nodes.py

```diff
@@ -181,92 +224,138 @@ def should_process_message_chunk(cls, message: AIMessageChunk) -> bool:
+    """
 
-            if not assistant_message:
-                raise ValueError("No memory message found.")
+    def run(self, state: AssistantState, config: RunnableConfig) -> PartialAssistantState:
+        human_message = find_last_message_of_type(state.messages, HumanMessage)
+        if not human_message:
+            raise ValueError("No human message found.")
 
-            # Compress the memory before saving it. The Perplexity's text is very verbose. It just complicates things for the memory collector.
+        core_memory, _ = CoreMemory.objects.get_or_create(team=self._team)
+        if human_message.content not in [
+            SCRAPING_CONFIRMATION_MESSAGE,
+            SCRAPING_REJECTION_MESSAGE,
+            ONBOARDING_INITIAL_MESSAGE,
+        ] and core_memory.initial_text.endswith("Answer:"):
+            # The user is answering to a question
+            core_memory.append_answer_to_initial_text(human_message.content)
+
+        if human_message.content == SCRAPING_REJECTION_MESSAGE:
+            core_memory.initial_text = ""
+            core_memory.save()
+
+        answers_left = core_memory.answers_left
+        if answers_left > 0:
             prompt = ChatPromptTemplate.from_messages(
-                [
-                    ("system", COMPRESSION_PROMPT),
-                    ("human", self._format_memory(assistant_message.content)),
-                ]
-            )
-            chain = prompt | self._model | StrOutputParser() | compressed_memory_parser
-            compressed_memory = cast(str, chain.invoke({}, config=config))
-            core_memory.set_core_memory(compressed_memory)
-        except:
-            core_memory.change_status_to_skipped()  # Ensure we don't leave the memory in a permanent pending state
-            raise
+                [("system", MEMORY_ONBOARDING_ENQUIRY_PROMPT)], template_format="mustache"
+            ).partial(core_memory=core_memory.initial_text, questions_left=answers_left)
 
-        return PartialAssistantState(
-            messages=[
-                AssistantMessage(
-                    content=SCRAPING_MEMORY_SAVED_MESSAGE,
-                    id=str(uuid4()),
-                )
-            ]
-        )
+            chain = prompt | self._model | StrOutputParser()
+            response = chain.invoke({}, config=config)
+
+            if "[Done]" not in response and "===" in response:
+                question = self._format_question(response)
+                core_memory.append_question_to_initial_text(question)
+                return PartialAssistantState(onboarding_question=question)
+        return PartialAssistantState(onboarding_question="")
 
     @property
     def _model(self):
-        return ChatOpenAI(model="gpt-4o-mini", temperature=0.3, disable_streaming=True, stop_sequences=["[Done]"])
+        return ChatOpenAI(model="o4-mini", temperature=0.3, disable_streaming=True, stop_sequences=["[Done]"])
```

Comment:
Does this work for you? When I run onboarding, this fails for me, with the message that only `temperature=1.0` is supported by `o4-mini`, and `stop_sequences` aren't supported at all

Reply:
I switched back to 4.1, o3-mini wasn't working but I forgot to remove it from the commit